staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1124,59 +1124,43 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 	if workersNumber < 1 {
 		workersNumber = 1
 	}
+
```

Comment:
This change looks correct to me, but I have one concern.

This effectively is creating a goroutine for every item in a list (which potentially may mean thousands or even more of goroutines). Starting too many goroutines quickly has caused us issues in the past, so if we decide to do that, I would like to see that:
- this actually improves something (some benchmark or sth)
- some explanation that starting too man goroutines at once is not a problem (golang folks were planning to work on that in the past so it's probably it's no longer really a problem at that scale)